### PR TITLE
refactor(components): [avatar] add namespace & instance type

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/avatar/src/avatar.vue [hunk 1]
```diff
@@ -31,24 +33,24 @@ export default defineComponent({
...
     const hasLoadError = ref(false)
 
     const avatarClass = computed(() => {
       const { size, icon, shape } = props
-      const classList = ['el-avatar']
-      if (size && typeof size === 'string') classList.push(`el-avatar--${size}`)
-      if (icon) classList.push('el-avatar--icon')
-      if (shape) classList.push(`el-avatar--${shape}`)
+      const classList = [ns.b()]
+      if (size && typeof size === 'string') classList.push(ns.m(size))
```
Comment (JeremyWuuuuu) [inline packages/components/avatar/src/avatar.vue:43]: Can we use `isString` from utils?

**Review by jw-foss — COMMENTED** [review]
Left a comment for you, thanks

File: packages/components/avatar/src/avatar.vue (hunk 1 above)
Comment (sxzz) [reply packages/components/avatar/src/avatar.vue]: Updated.

**Review by jw-foss — APPROVED** [review]
LGTM, approved

**Pull request merged**